### PR TITLE
Add ability to skip certificate verification during service checks

### PR DESCRIPTION
**Pull request by LetTheAwesomeBegin**
Fixes gh-261

### CODE REVIEW

File: spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoServiceRegistrationCustomizedDiscoveryHealthCheckTlsSkipVerifyTests.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT;
+
+/**
+ * @author Patrick Hi
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = ConsulAutoServiceRegistrationCustomizedDiscoveryHealthCheckTlsSkipVerifyTests.TestConfig.class,
+	properties = { "spring.application.name=myTestService-DiscoveryHealthCheckTlsSkipVerify", },
+		webEnvironment = DEFINED_PORT)
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoServiceRegistrationCustomizedDiscoveryHealthCheckTlsSkipVerifyTests.java]: Should be random.

File: spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoServiceRegistrationCustomizedDiscoveryHealthCheckTlsSkipVerifyTests.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public class ConsulAutoServiceRegistrationCustomizedDiscoveryHealthCheckTlsSkipVerifyTests {
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoServiceRegistrationCustomizedDiscoveryHealthCheckTlsSkipVerifyTests.java]: Change to `ConsulAutoRegistrationHealthCheckTlsSkipVerifyTests`

**Comment by antinodes** [comment]
Hi.  This PR fixes a problem I am experiencing with one of my projects.  If there are additional changes that are needed before this PR can be merged is there some way I can help out?

File: spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistrationHealthCheckTlsSkipVerifyTests.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	@Autowired
+	private ConsulAutoRegistration registration;
+
+	@Autowired
+	private ConsulDiscoveryProperties properties;
+
+	@BeforeClass
+	public static void initClass() {
+		System.setProperty("spring.cloud.consul.discovery.health-check-tls-skip-verify", "true");
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistrationHealthCheckTlsSkipVerifyTests.java]: This should go in the `properties` attr of `@SpringBootTest`

**Pull request merged**